src/backend/distributed/utils/maintenanced.c

```diff
@@ -267,9 +268,24 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		 * to properly add it to the hash.
 		 * Alternatively, don't continue if another worker exists.
 		 */
-		proc_exit(0);
+		bool immediatelyExit = true;
+		if (myDbData && myDbData->workerPid != 0)
+		{
```

Comment:
I don't see a huge problem with this, but I'm wondering whether this is still relevant and might confuse future readers. 

The !myDbData case covers the restart-on-crash scenario where the shared memory is gone. 
The workerPid != 0 scenario used to cover restart-on-error scenarios, which are now handled by before_shmem_exit.

I don't believe there is a scenario in which `myDbData && myDbData->workerPid != 0` can be true now, or even a scenario in which `myDbData->workerPid != 0` can be true. However, I might be wrong.

I'm wondering whether we should change to something like:
```c
if (myDbData && myDbData->workerPid != 0 && kill(myDbData->workerPid, 0) != 0)
{
    /* 
     * The last worker process not longer exists, but failed to clean up. Set
     * daemonStarted to false to create a new daemon from the next session that
     * calls InitializeMaintenanceDaemonBackend.
     */
    myDbData->daemonStarted = false;
    myDbData->workerPid = 0;
}

proc_exit(0);
```

I'll leave it up to you to decide. I'm ok with all 3 options (reverting, leaving as is, or `daemonStarted = false`).

Reply:
Removed `workerPid != 0` in favor of `Assert(myDbData->workerPid == 0)`